### ISSUE TITLE
node version

### ISSUE THREAD

**Issue by mrhouzlane** [trufflesuite/truffle#5498]
- [ ] I've [opened a support ticket](https://trufflesuite.zendesk.com/hc/en-us/requests/new) before filing this issue.

---------------------------

## Issue

Node version. 

## Steps to Reproduce

Run : ` truffle develop`

## Expected Behavior

 Expected to open console 

## Actual Results

``` Error: Cannot find module './uws_darwin_x64_108.node'
Falling back to a NodeJS implementation; performance may be degraded.
Error: error:0308010C:digital envelope routines::unsupported 
```

It worked when I used node v16.15.0  (doing ```nvm use 16```  ) 

## Environment

* Operating System:  macOs v 12.0.1
* Truffle version (`truffle version`): Truffle v5.5.15 (core: 5.5.15)
* node version (`node --version`):  18.4.0
* npm version (`npm --version`):  8.12.1

**Comment by Hitanshu5979** [comment]
Truffle currently only supports Node.js till the last stable release, ie, 16.17.0 LTS.
Downgrading from the Beta Version, ie, 18.4.0 seems to be the most straightforward solution to this problem.

**Comment by lsqproduction** [comment]
Hi, @mrhouzlane were you able to downgrade to node version 16 and able to have fun with `truffle develop`?

**Comment by guilhermemauro** [comment]
Hi guys, I got run using Node v18.9.0 setting this env before run truffle:

`export NODE_OPTIONS=--openssl-legacy-provider`

$ truffle --version:
Truffle v5.5.29 (core: 5.5.29)
Ganache v7.4.0
Solidity v0.5.16 (solc-js)
Node v18.9.0
Web3.js v1.7.4

**Comment by cliffoo** [comment]
We're currently working hard on node 18 support, you can track it on [pr#5543](https://github.com/trufflesuite/truffle/pull/5543).

For now you can export that openssl node option but please keep in mind that it isn't fully tested.

**Comment by eggplantzzz** [comment]
This should be fixed in the latest Truffle (v5.6.0).
